### PR TITLE
Externals: Add soundtouch

### PR DESCRIPTION
**Pull request by MerryMage**
This PR just contains SoundTouch. Sumbitted in preparation for another audio-related PR.

### CODE REVIEW

**Comment by JayFoxRox** [comment]
(direct quote from CONTRIBUTING.md:)

* Don't ever introduce new external dependencies into Core

I do realize we probably need something like this to avoid code bloat on our end, but externals can be a huge PITA to maintain (which is probably why the quote above exists - unless it means using them without adding them to the repo which would be.. well.. worse, but I'm getting off-topic here).

Especially with this being a subdirectory and not a git submodule I'm slightly worried about maintenance.
With that in mind I'd like to know if you looked at any other libraries (possibly slimmer?) and also the purpose of the library will be in Citra. Will it be used for effects by the DSP or is it purely for time-scaling of audio in case of host <=> guest sync / perf issues?

**Comment by MerryMage** [comment]
> is it purely for time-scaling of audio in case of host <=> guest sync / perf issues

Purely the latter. I've looked at and considered one other option ([rubberband](http://breakfastquay.com/rubberband/)) which actually uses an algorithm I prefer and can be easily submoduled, but both audio quality and performance turn out to be abysmal for the frame-by-frame adjustment of tempo Citra requires.

> externals can be a huge PITA to maintain

Would me mirroring my own git repo be perferrable?

**Comment by MerryMage** [comment]
Updated to submodule soundtouch instead.

**Comment by bunnei** [comment]
LGTM

File: .gitmodules [hunk 1]
```diff
@@ -7,3 +7,6 @@
 [submodule "nihstro"]
 	path = externals/nihstro
 	url = https://github.com/neobrain/nihstro.git
+[submodule "soundtouch"]
+	path = externals/soundtouch
+	url = https://github.com/citra-emu/soundtouch.git
```
Comment (JayFoxRox) [inline .gitmodules]: I've already said it on IRC (unless my message was dropped) before the merge: Our standard naming scheme for external submodules is `citra-emu/ext-<name>.git`.
We should rename the soundtouch module for consistency.

**Pull request merged**